src/test/java/com/karankumar/bookproject/backend/util/DateUtilsTest.java

```diff
@@ -78,6 +87,28 @@ void correctlyGetWeeksLeftInYearFromCurrentWeek() {
                 .isEqualTo(calculateWeeksLeftInYear(mockLastWeekOfYear()));
     }
 
+    @Test
+    void correctlyCheckIfDateIsInCurrentYear() {
+        LocalDate lastDayOfYear = LocalDate.of(MOCK_CURRENT_DATE.getYear(), 12, 31);
+
+        assertThat(DateUtils.dateIsInCurrentYear(MOCK_CURRENT_DATE)).isTrue();
+        assertThat(DateUtils.dateIsInCurrentYear(MOCK_CURRENT_DATE.minusDays(1))).isTrue();
+        assertThat(DateUtils.dateIsInCurrentYear(lastDayOfYear)).isTrue();
+
+        assertThat(DateUtils.dateIsInCurrentYear(MOCK_CURRENT_DATE.minusYears(1))).isFalse();
+        assertThat(DateUtils.dateIsInCurrentYear(MOCK_CURRENT_DATE.plusYears(1))).isFalse();
+    }
+
+    @Test
+    void correctlyCheckIfDateIsInFuture() {
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE.plusYears(1))).isTrue();
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE.plusDays(1))).isTrue();
+
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE)).isFalse();
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE.minusDays(1))).isFalse();
+        assertThat(DateUtils.isDateInFuture(MOCK_CURRENT_DATE.minusMonths(1))).isFalse();
```

Comment:
Could you please use `assertSoftly()` here?

Reply:
Good idea, updated.